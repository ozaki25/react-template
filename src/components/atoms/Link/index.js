import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

function Link({ children, onClick }) {
  return <LinkText onClick={onClick}>{children}</LinkText>;
}

export const LinkText = Button.extend`
  color: #666;
  font-size: 16px;
`;

Link.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
};

Link.defaultProps = {
  onClick: null,
};

export default Link;
