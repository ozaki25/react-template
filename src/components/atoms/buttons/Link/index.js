import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Color from 'src/constants/Color';

const Link = ({ children, onClick }) => <Button onClick={onClick}>{children}</Button>;

const Button = styled.button`
  color: ${Color.link};
  font-size: 16px;
  text-decoration: underline;
`;

Link.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
};

Link.defaultProps = {
  onClick: null,
};

export default Link;
