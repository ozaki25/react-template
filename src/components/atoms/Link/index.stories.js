import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Link from '.';

const stories = storiesOf('atoms', module);

stories.add(
  'Link',
  withInfo('リンク用文字列')(() => (
    <Link onClick={action('clicked Link')}>{text('text', 'リンク')}</Link>
  )),
);
