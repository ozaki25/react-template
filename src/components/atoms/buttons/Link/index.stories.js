import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Link from '.';

const stories = storiesOf('atoms/buttons', module);

stories.add(
  'Link',
  withInfo('リンク')(() => <Link onClick={action('clicked Link')}>{text('text', 'リンク')}</Link>),
);
