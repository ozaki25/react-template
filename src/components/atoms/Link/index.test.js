import React from 'react';
import { snapshot } from '../../../../test/helpers';
import Link, { LinkText } from './';

const props = {
  onClick: jest.fn(),
};

snapshot('Link', <Link {...props}>リンク</Link>);
snapshot('Link/LinkText', <LinkText>リンク</LinkText>);
