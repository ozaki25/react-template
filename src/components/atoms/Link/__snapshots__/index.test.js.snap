// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Link 1`] = `
<styled.button
  onClick={[Function]}
>
  リンク
</styled.button>
`;

exports[`should match snapshot: Link/LinkText 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 0;
  color: #666;
  font-size: 16px;
}

.c0:active {
  opacity: 0.5;
}

<button
  className="c0"
>
  リンク
</button>
`;
